packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts

```diff
@@ -164,50 +179,59 @@ export class SOQLEditorInstance {
         this.updateTextDocument(this.document, soql);
         break;
       }
+      case MessageType.UI_TELEMETRY: {
+        const { unsupported } = e.payload as TelemetryModelJson;
+        const hasUnsupported = Array.isArray(unsupported)
+          ? unsupported.length
+          : unsupported;
+        if (hasUnsupported) {
+          trackErrorWithTelemetry(
+            'syntax_unsupported',
+            JSON.stringify(e.payload)
+          ).catch(console.error);
+          const message = nls.localize('info_syntax_unsupported');
+          channelService.appendLine(message);
+        }
+        break;
+      }
       case MessageType.SOBJECT_METADATA_REQUEST: {
-        this.retrieveSObject(e.payload as string).catch(() => {
-          channelService.appendLine(
-            `An error occurred while handling a request for object metadata for the ${e.payload} object.`
-          );
-        });
+        this.retrieveSObject(e.payload as string);
         break;
       }
       case MessageType.SOBJECTS_REQUEST: {
-        this.retrieveSObjects().catch(() => {
-          channelService.appendLine(
-            `An error occurred while handling a request for object names.`
-          );
-        });
+        this.retrieveSObjects();
         break;
       }
       case MessageType.RUN_SOQL_QUERY: {
-        this.handleRunQuery().catch(() => {
-          channelService.appendLine(
-            `An error occurred while running the SOQL query.`
-          );
-        });
+        this.handleRunQuery();
         break;
       }
       default: {
-        console.log('message type is not supported');
+        const message = nls.localize('error_unknown_error', e.type);
+        channelService.appendLine(message);
+        trackErrorWithTelemetry('message_unknown', e.type).catch(console.error);
       }
     }
   }
 
   protected handleRunQuery(): Promise<void> {
     // Check to see if a default org is set.
     if (!workspaceContext.username) {
-      // i18n
-      const message = `No default org found. Set a default org to use SOQL Builder. Run "SFDX: Create a Default Scratch Org" or "SFDX: Authorize an Org" to set one.`;
+      const message = nls.localize('info_no_default_org');
```

Comment:
I'm not sure if both channel and poptarts are both necessary here.  But it sure gets the point across.

<img width="473" alt="no-default-org" src="https://user-images.githubusercontent.com/599418/101825847-70da1c80-3aeb-11eb-8894-94de9d7765d7.png">


<img width="1001" alt="no-default-org-channel" src="https://user-images.githubusercontent.com/599418/101825827-6a4ba500-3aeb-11eb-9c94-85fa4c33983e.png">